server/serverMethods.js

```diff
@@ -619,51 +628,69 @@ function ServerMethods(aLogLevel, aModules) {
   function postRoomDial(aReq, aRes) {
     var tbConfig = aReq.tbConfig;
     var roomName = aReq.params.roomName.toLowerCase();
-    var tbConfig = aReq.tbConfig;
     var body = aReq.body;
     if (!body || !body.phoneNumber) {
       logger.log('postRoomDial => missing body parameter: ', aReq.body);
       aRes.status(400).send(new ErrorInfo(100, 'Missing required parameter'));
       return;
     }
-    var phoneNumber = body.phoneNumber;
-    if (dialedNumbersInfo[phoneNumber]) {
-      return;
-    }
-    var otInstance = tbConfig.otInstance;
 
-    serverPersistence
-      .getKey(redisRoomPrefix + roomName)
-      .then((sessionInfo) => {
-        const sessionId = JSON.parse(sessionInfo).sessionId;
-        const token = tbConfig.otInstance.generateToken(sessionId, {
-          role: 'publisher',
-          data: '{"sip":true, "role":"client", "name":"' + phoneNumber + '"}'
-        });
-        var options = {
-          // Plivo accepts custom headers that start with 'X-PH'
-          headers: {
-            'X-PH-ROOMNAME': roomName,
-            'X-PH-DIALOUT-NUMBER': phoneNumber
-          },
-          auth: {
-            username: tbConfig.sipUsername,
-            password: tbConfig.sipPassword
-          },
-          secure: true
+    var auth = new GoogleAuth; // eslint-disable-line new-parens
+    var client = new auth.OAuth2(tbConfig.googleId, '', '');
+    var verifyIdTokenPromise = Utils.promisify(client.verifyIdToken, 1, client);
+
+    verifyIdTokenPromise(body.googleIdToken, tbConfig.googleId)
+      .then((login) => {
+        var payload = login.getPayload();
+        var domain = payload.hd;
+        if (domain !== tbConfig.googleHostedDomain) {
+          logger.log('postRoomDial => authenticated token domain did not match config: ', domain);
+          aRes.status(403).send(new ErrorInfo(100, 'Authenticaton Domain Invalid'));
+        } else {
+          var phoneNumber = body.phoneNumber;
+          if (dialedNumbersInfo[phoneNumber]) {
+            return;
```

Comment:
If you don't do a aRes.status(something).send(somethingelse) here the client is going to 'hang' till it gets a timeout. Basically the connection is kept open till you send something.